lib/ide/parser/src/api.rs

```diff
@@ -15,9 +59,14 @@ pub use ast::Ast;
 pub trait IsParser : Debug {
     /// Parse program.
     fn parse(&mut self, program:String, ids:IdMap) -> Result<Ast>;
+
+    /// Parse a module content that contains idmap and metadata.
+    fn parse_with_metadata
+    (&mut self, program:String) -> Result<ModuleWithMetadata>;
```

Comment:
What is this `Result`? Sometimes you are using `std::result::Result`, sometimes `api::Result` (both are OK), and now also `Result` - this is inconsistent.

Reply:
This is api::Result because we are in api.